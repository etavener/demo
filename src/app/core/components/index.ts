export * from './card-form/card-form.component';
export * from './completed-message/completed-message.component';
export * from './payment-info/payment-info.component';
export * from './progress-display/progress-display.component';
export * from './input-expiry-date/input-expiry-date.component';
