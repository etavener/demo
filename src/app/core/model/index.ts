export * from './payment';
export * from './worldpayResponse';
