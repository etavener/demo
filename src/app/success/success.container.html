<div class="payment-success">
  <mat-card>
    <mat-card-title>
      <h1>Success</h1>
    </mat-card-title>
    <mat-card-content>
      <completed-message></completed-message>
    </mat-card-content>
  </mat-card>
</div>
