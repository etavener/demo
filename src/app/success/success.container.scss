.payment-success {

  max-width: 500px;
}
