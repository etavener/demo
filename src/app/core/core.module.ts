import { NgModule } from '@angular/core';
import {
  CardFormComponent,
  PaymentInfoComponent,
  CompletedMessageComponent,
  ProgressDisplayComponent,
  InputExpiryDateComponent,
} from './components';

import {
  DateAdapter, MAT_DATE_FORMATS, MAT_DATE_LOCALE,
  MatButtonModule,
  MatDatepickerModule,
  MatDividerModule,
  MatFormFieldModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatSelectModule,
  MatStepperModule
} from '@angular/material';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule} from '@angular/router';
import { STEPPER_GLOBAL_OPTIONS } from '@angular/cdk/stepper';
import { HttpClientModule } from '@angular/common/http';
import { MomentDateAdapter } from '@angular/material-moment-adapter';
import {WorldpayService} from './services/worldpay.service';
import {BrowserAnimationsModule} from '@angular/platform-browser/animations';

export const MY_FORMATS = {
  parse: {
    dateInput: 'MM/YYYY',
  },
  display: {
    dateInput: 'MM/YYYY',
    monthYearLabel: 'MMM YYYY',
    dateA11yLabel: 'LL',
    monthYearA11yLabel: 'MMMM YYYY',
  },
};

const DECLARATIONS = [
  CardFormComponent,
  PaymentInfoComponent,
  CompletedMessageComponent,
  ProgressDisplayComponent,
  InputExpiryDateComponent
];

@NgModule({
  declarations: [
    ...DECLARATIONS
  ],
  exports: [
    ...DECLARATIONS
  ],
  imports: [
    FormsModule,
    ReactiveFormsModule,
    RouterModule,
    MatStepperModule,
    MatIconModule,
    MatDatepickerModule,
    MatInputModule,
    MatFormFieldModule,
    MatProgressSpinnerModule,
    MatSelectModule,
    MatButtonModule,
    MatDividerModule,
    HttpClientModule,
    BrowserAnimationsModule
  ],
  providers: [
    WorldpayService,
    {
      provide: STEPPER_GLOBAL_OPTIONS,
      useValue: { displayDefaultIndicatorType: false }
    },
    {
      provide: DateAdapter,
      useClass: MomentDateAdapter,
      deps: [MAT_DATE_LOCALE]},
    {
      provide: MAT_DATE_FORMATS,
      useValue: MY_FORMATS
    },
  ]
})
export class CoreModule { }
