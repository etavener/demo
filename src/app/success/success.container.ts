import { Component } from '@angular/core';

@Component({
  selector: 'success-container',
  templateUrl: './success.container.html',
  styleUrls: ['./success.container.scss']
})
export class SuccessContainer {

  constructor() {}

}
