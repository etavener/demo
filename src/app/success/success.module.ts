import { NgModule } from '@angular/core';
import { SuccessContainer } from './success.container';
import { CoreModule } from '../core/core.module';
import {MatCardModule} from '@angular/material';

@NgModule({
  declarations: [
    SuccessContainer
  ],
  imports: [
    CoreModule,
    MatCardModule
  ],
  providers: []
})
export class SuccessModule { }
