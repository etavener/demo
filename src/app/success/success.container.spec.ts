import { async, ComponentFixture, TestBed, } from '@angular/core/testing';
import { SuccessContainer } from './success.container';
import { SuccessModule } from './success.module';

describe('Success container', () => {
  let component: SuccessContainer;
  let fixture: ComponentFixture<SuccessContainer>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [],
      imports: [
        SuccessModule
      ]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SuccessContainer);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should render the component', () => {
    expect(fixture).toMatchSnapshot();
  });
});
