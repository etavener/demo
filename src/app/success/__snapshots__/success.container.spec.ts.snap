// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success container should render the component 1`] = `
<success-container>
  <div
    class="payment-success"
  >
    <mat-card
      class="mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        <h1>
          Success
        </h1>
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <completed-message>
          <div>
            <div>
              Your payment has now been authorized.
            </div>
          </div>
        </completed-message>
      </mat-card-content>
    </mat-card>
  </div>
</success-container>
`;
